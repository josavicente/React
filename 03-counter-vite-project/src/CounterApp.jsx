import { useState } from 'react';
import PropTypes from 'prop-types';


export const CounterApp = ( { value } ) => {


      const [ counter, setCounter] = useState(0);

      const handleAdd = () => {
            // setCounter (counter + 1);
            setCounter( (c) => c + 1)
      };

      const handleSubstract = () => {
            // setCounter (counter - 1);
            setCounter( (c) =>  c - 1 )
      };

      const handleReset = () => {
            // setCounter (0);
            setCounter( (c) => 0)
      };

      return ( 
      <>
            <h1>CounterApp</h1>
            <h2> { counter } </h2>
            {/* <button onClick = { (event) => handleEvent(event) }>
                  +1
            </button> */}
            <button onClick = { handleAdd }>
                  +1
            </button>
            <button onClick = { handleSubstract }>
                  -1
            </button>
            <button onClick = { handleReset }>
                  Reset
            </button>

      </>
)};


CounterApp.propTypes = {
      value: PropTypes.number
};

CounterApp.defaultProps = {
      value: 100
}