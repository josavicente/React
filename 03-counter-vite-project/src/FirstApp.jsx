
// const getResult = () => {
//       return 'Josa';
// }
import PropTypes from 'prop-types';

export const FirstApp = ( {title, message} ) => {

      return ( 
      <>
      <h1 data-testid="test-title">{title}</h1>
      <p>{message}</p>
      </>
)};

FirstApp.propTypes = {
      title: PropTypes.string.isRequired,
      message: PropTypes.string
};

FirstApp.defaultProps = {
      // title: 'No hay t√≠tulo',
      message: '666'
}
