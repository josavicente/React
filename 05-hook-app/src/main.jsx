import React from 'react'
import ReactDOM from 'react-dom/client'
// import { CounterApp } from './01-useState/CounterApp'
// import { CounterWithCustomHook } from './01-useState/counterWithCounterHook'
// import { FormWithCustomHook } from './02-useEffect/FormWithCustomHook'
// import { SimpleForm } from './02-useEffect/simpleForm'
// import { MultipleCustomHooks } from './03-examples/MultipleCustomHooks'
// import { FocusScreen } from './04-useRef/FocusScreen'
// import { Layout } from './05-useLayoutEffect/Layout'
// import { Memorize } from './06-memos/memorize'
import { MemorizeHook } from './06-memos/MemorizeHook'
// import { HooksApp } from './HooksApp'
import './index.css'

ReactDOM.createRoot(document.getElementById('root')).render(
  <React.StrictMode>
    {/* <CounterApp /> */}
    {/* <CounterWithCustomHook /> */}
    {/* <FormWithCustomHook/> */}
    {/* <MultipleCustomHooks /> */}
    {/* <FocusScreen /> */}
    {/* <Layout /> */}
    {/* <Memorize /> */}
    <MemorizeHook />
  </React.StrictMode>
)
