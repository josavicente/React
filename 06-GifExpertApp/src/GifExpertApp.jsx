

//  const apiKey = '0SQ78YHAvJ2yMgLzOGGpwnMJ0N8cV95t';
//  const resp   = await fetch(`http://api.giphy.com/v1/gifs/random?api_key=${ apiKey }`);
//  const { data } = await resp.json(); 

export const GifExpertApp = () => {
      return (
            <>
            {/* TÃ­tulo */}
            <h1>GifExpertApp</h1>

            {/* Entrada de datos */}

            {/* Listado de Gif's */}
                  {/* GifItem */}
            </>
      )
};