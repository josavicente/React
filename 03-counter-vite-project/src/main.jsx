import React from 'react';
import ReactDOM from 'react-dom/client';
import { FirstApp } from './FirstApp';
import { HelloWorldApp, } from './HelloWorldApp';
import './styles.css';


ReactDOM.createRoot(document.getElementById('root')).render(
      <React.StrictMode>
            <FirstApp  />
      </React.StrictMode>
);