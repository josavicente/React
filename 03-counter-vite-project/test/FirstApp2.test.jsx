import { FirstApp } from "../src/FirstApp";
import { render, screen } from '@testing-library/react';

describe('Pruebas en FirstApp', () => { 
      
      const title = 'Hola soy Goku';
      const message = '666';

      test('Debe de hacer match con el snapshot', () => {
            
            const { container } = render(<FirstApp title= { title }/>)
            expect( container ).toMatchSnapshot();
      });

      test('Debe de mostrar el mensaje Hola soy Goku', () => {
            render(<FirstApp title= { title }/>)
            expect( screen.getByText(title)).toBeTruthy();
      })
      test('Debe de mostrar el tÃ­tulo en un h1', () => {

            render(<FirstApp title= { title }/>)
            expect(screen.getByRole('heading', { level: 1 }).innerHTML).toContain(title)

            // const { container, getByText, getByTestId } = 
            // expect( getByText(title) ).toBeTruthy();
            // expect(getByTestId('test-title')).toBeTruthy();
            // expect(getByTestId('test-title').innerHTML).toContain(title);
      }) 

      test('Debe de mostrar el message enviado por props', () => {

            render(
                  <FirstApp 
                        title= { title }
                        message={ message }
                  />)
            // expect( getByText(message) ).toBeTruthy();

            // // const h1 = container.querySelector('h1');
            // // console.log(h1.innerHTML);
            // // expect(h1.innerHTML).toContain(title);
            // // expect(getByTestId('test-title')).toBeTruthy();
            // // expect(getByTestId('test-title').innerHTML).toContain(title);
            expect (screen.getAllByText(message).length).toBe(1)
      })
})
