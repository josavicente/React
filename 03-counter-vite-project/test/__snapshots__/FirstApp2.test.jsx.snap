// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en FirstApp Debe de hacer match con el snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Hola soy Goku
  </h1>
  <p>
    666
  </p>
</div>
`;
